app/src/main/java/org/jellyfin/androidtv/data/eventhandling/SocketHandler.kt

```diff
@@ -0,0 +1,164 @@
+package org.jellyfin.androidtv.data.eventhandling
+
+import android.content.Context
+import android.widget.Toast
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.data.model.DataRefreshService
+import org.jellyfin.androidtv.ui.playback.MediaManager
+import org.jellyfin.androidtv.util.apiclient.PlaybackHelper
+import org.jellyfin.apiclient.model.entities.MediaType
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.sessionApi
+import org.jellyfin.sdk.api.client.extensions.userLibraryApi
+import org.jellyfin.sdk.api.sockets.SocketInstance
+import org.jellyfin.sdk.api.sockets.addGeneralCommandsListener
+import org.jellyfin.sdk.api.sockets.addListener
+import org.jellyfin.sdk.model.api.GeneralCommandType
+import org.jellyfin.sdk.model.api.LibraryUpdateInfo
+import org.jellyfin.sdk.model.api.PlaystateCommand
+import org.jellyfin.sdk.model.extensions.getValue
+import org.jellyfin.sdk.model.serializer.toUUIDOrNull
+import org.jellyfin.sdk.model.socket.LibraryChangedMessage
+import org.jellyfin.sdk.model.socket.PlayMessage
+import org.jellyfin.sdk.model.socket.PlayStateMessage
+import timber.log.Timber
+import java.util.UUID
+
+class SocketHandler(
+	private val context: Context,
+	private val api: ApiClient,
+	private val dataRefreshService: DataRefreshService,
+	private val mediaManager: MediaManager,
+) {
+	private val coroutineScope = CoroutineScope(Dispatchers.IO)
+	private var socketInstance: SocketInstance? = null
+
+	suspend fun updateSession() {
+		api.sessionApi.postCapabilities(
+			playableMediaTypes = listOf(MediaType.Video, MediaType.Audio),
+			supportsMediaControl = true,
+			supportedCommands = listOf(
+				GeneralCommandType.DISPLAY_MESSAGE,
+				GeneralCommandType.SEND_STRING,
+			),
+		)
+
+		if (socketInstance != null) socketInstance?.updateCredentials()
+		else socketInstance = createInstance()
+	}
+
+	private fun createInstance() = api.ws().apply {
+		// Library
+		addListener<LibraryChangedMessage> { message -> onLibraryChanged(message.info) }
+
+		// Media playback
+		addListener<PlayMessage> { message -> onPlayMessage(message) }
+		addListener<PlayStateMessage> { message -> onPlayStateMessage(message) }
+
+		// General commands
+		addGeneralCommandsListener(setOf(GeneralCommandType.DISPLAY_CONTENT)) { message ->
+			val itemId by message
+			val itemUuid = itemId?.toUUIDOrNull()
+
+			if (itemUuid != null) onDisplayContent(itemUuid)
+		}
+		addGeneralCommandsListener(setOf(GeneralCommandType.DISPLAY_MESSAGE, GeneralCommandType.SEND_STRING)) { message ->
+			val header by message
```

Comment:
messages send from here work:
<img width="366" alt="Screen Shot 2022-03-27 at 5 01 05 AM" src="https://user-images.githubusercontent.com/28824542/160280517-cbd0a2df-acca-4f13-aa9e-2a575984e8cd.png">

messages sent from either of these two just show "null":
<img width="1565" alt="Screen Shot 2022-03-27 at 5 00 46 AM" src="https://user-images.githubusercontent.com/28824542/160280534-4cbad581-2494-41c2-85bf-90c4af3015bb.png">

It looks like both the message sent from the "active devices" section (top screenshot) and the top one in the remote control thing are `DisplayMessage` commands, and the bottom one in the remote control is `SendString`

Reply:
Unable to reproduce with 10.7.7. Can you post the raw messages from the logs?

```
I/KtorSocketInstanceConnection: Receiving (raw) message {"MessageType":"GeneralCommand","MessageId":"ea344f1a82ce46e9bdf8bac4672b81f8","Data":{"Name":"DisplayMessage","ControllingUserId":"8f25a6f142e14d748aff1b041030128f","Arguments":{"Header":"a","Text":"b"}}}
```
```
I/KtorSocketInstanceConnection: Receiving (raw) message {"MessageType":"GeneralCommand","MessageId":"682e6b1b016a47608709c4821811277b","Data":{"Name":"SendString","ControllingUserId":"8f25a6f142e14d748aff1b041030128f","Arguments":{"String":"test2"}}}
```

Reply:
dashboard one:

2022-03-27 05:50:23.358 8679-13240/org.jellyfin.androidtv.debug I/KtorSocketInstanceConnection: Receiving (raw) message {"MessageType":"GeneralCommand","MessageId":"a3bb43622f9f4111b86ace60f58936ef","Data":{"Name":"DisplayMessage","ControllingUserId":"00000000000000000000000000000000","Arguments":{"Header":"Message from Server","Text":"test message","TimeoutMs":"5000"}}}

(I didn't replace the ControllingUserId value with zeros, that's how it is in the log)

remote control menu top one:

`8679-8771/org.jellyfin.androidtv.debug I/KtorSocketInstanceConnection: Receiving (raw) message {"MessageType":"GeneralCommand","MessageId":"be8a46df6df2484f976766f6a8b3c49c","Data":{"Name":"DisplayMessage","ControllingUserId":"MYUSERID","Arguments":{}}}`

bottom one:

2022-03-27 05:49:03.936 8679-8771/org.jellyfin.androidtv.debug I/KtorSocketInstanceConnection: Receiving (raw) message {"MessageType":"GeneralCommand","MessageId":"86025bac44b745aa9e0bbb35e177702f","Data":{"Name":"SendString","ControllingUserId":"MYUSERID","Arguments":{}}}

Reply:
Looks like an issue in jellyfin-web or the server with 10.8.